### PR TITLE
Trigger animation on success of delete

### PR DESCRIPTION
**Pull request by cizil-keyvalue**

### CODE REVIEW

File: src/components/NotificationCard.vue [hunk 1]
```diff
@@ -60,14 +62,19 @@ let cardContainerStyle: CSSProperties;
 
 const defaultAvatar: string = props?.darkMode ? defaultAvatarDark : defaultAvatarLight;
 
-const handleDelete = (event: MouseEvent | KeyboardEvent) => {
-  deleteAnimation.value = 'siren-sdk-delete-animation';
+const handleDelete = async (event: MouseEvent | KeyboardEvent): Promise<void> => {
+  event.stopPropagation();
+  const isSuccess = await props.deleteById(props.notification.id, false);
```
Comment (nikhin-keyvalue) [inline src/components/NotificationCard.vue:67]: what is the value of isSuccess when API fail ?

File: src/components/SirenPanel.vue [hunk 2]
```diff
@@ -142,15 +142,16 @@ const triggerOnError = computed(
...
+    const response = await deleteById(id, shouldUpdateList);
 
+    if (response?.data) isSuccess = true;
     if (response) triggerOnError.value(response);
-  } catch (err) {
-    // TODO: handle error
-  }
-});
+
+  return isSuccess;
```
Comment (nikhin-keyvalue) [inline src/components/SirenPanel.vue]: const response = await deleteById(id, shouldUpdateList);
  if (response) triggerOnError.value(response);
  return !!response?.data;

WIll this work ?

File: src/composables/useSiren.ts [hunk 3]
```diff
@@ -46,12 +46,12 @@ const useSiren = () => {
...
 
-  const deleteById = async (id: string) => {
+  const deleteById = async (id: string, shouldUpdateList: boolean = true) => {
     if (siren.value)
       if (id?.length > 0) {
         const response = await siren.value?.deleteById(id);
 
-        if (response && response.data) {
+        if (response && response.data && shouldUpdateList) {
           const payload = { id, action: eventTypes.DELETE_ITEM };
```
Comment (nikhin-keyvalue) [inline src/composables/useSiren.ts:55]: if (response?.data && shouldUpdateList) {

**Review by nikhin-keyvalue — CHANGES_REQUESTED** [review]

File: src/components/NotificationCard.vue (hunk 1 above)
Comment (cizil-keyvalue) [reply src/components/NotificationCard.vue:67]: deleteById will return false and will trigger onError

**Review by nikhin-keyvalue — APPROVED** [review]

**Pull request merged**